_posts/2023-09-20-intro-to-the-various-java-based-k8s-oc-tools.adoc

```diff
@@ -0,0 +1,607 @@
+simple `Dockerfile` based on the project used frameworks(`wildfly-jar`
+or Quarkus for
+examplefootnote:[https://github.com/eclipse/jkube/tree/master/jkube-kit[jkube/jkube-kit
+at master · eclipse/jkube]]). In addition, the JKube can generate k8s
+manifest files and do the k8s/openshift deployments. I have written a
+blog post showing its
+usagefootnote:[https://weinan.io/2023/06/23/jkube.html[Using JKube To Do
+Kubernetes Deployment]].
+
+Comparing with other tools, the JKube can do various tasks, which is
+very convenient. Nevertheless other tools may focus on one task and
+provide some more fine grain controls.
+
+== https://github.com/fabric8io/docker-maven-plugin[Fabric8 Docker Maven plugin]
+
+As already introduced in above Eclipse JKube section, the
+https://github.com/fabric8io/docker-maven-plugin[Fabric8 Docker Maven
+plugin] is part of the Fabric8 ecosystem. This project is focusing on
+the Docker image related tasks. It can define a docker image in its
+maven plugin configuration without having a `Dockerfile`, and it can do
+the container deployment/undeployment during the Maven integration test
+phase. We will see this plugin usage later.
+
+== https://github.com/fabric8io/kubernetes-client[Fabric8 Kubernetes client]
+
+As the name suggests, this project provides Java based client to
+interact with the Kubernetes and OpenShift services. The functions of
+this Java based client is similar to the `kubectl` and `oc` commands. So
+this tool is convenient to use for writing k8s and OpenShift related
+tests.
+
+== https://github.com/GoogleContainerTools/jib[jib]
+
+`jib` is a tool provided by Google and it can be used to build images
+without Docker daemon installed, so it can produce a docker image
+standalone.
+
+After introducing the above tools, I’ll give an introduction to see
+their usages in some WildFly related projects as examples.
+
+== Usage of the above tools in WildFly related projects
+
+Firstly we can discuss about Quarkus. The Quarkus has its
+`Kubernentes Extension`footnote:[https://quarkus.io/guides/deploying-to-kubernetes[Quarkus
+Kubernetes Extension]], which provides these components:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-kubernetes</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-container-image-jib</artifactId>
+</dependency>
+----
+
+As the components it provides as shown above, the `quarkus-kubernetes`
+is based on Dekorate, and it can be used to generate k8s manifest file
+for Quarkus based projects. The `quarkus-container-image-jib` is based
+on the `jib`, and this extension is used to build docker image for the
+Quarkus based projects.
+
+Next we can see the
+`wildfly-cloud-tests`footnote:[https://github.com/wildfly-extras/wildfly-cloud-tests[WildFly
+Cloud Testsuite]] project. This project is still under development and
+does not have a formal release yet. Nevertheless it’s still a good
```

Comment:
FWIW I plan to do a release this week. It will likely be an Alpha or simply a 0.1.0 type of release though.